PocketKit/Sources/PocketKit/Navigation/NavigationSidebarViewController.swift

```diff
@@ -0,0 +1,98 @@
+        var configuration = UICollectionLayoutListConfiguration(appearance: .insetGrouped)
+        configuration.headerMode = .none
+        configuration.backgroundColor = UIColor(.ui.white1)
+        return UICollectionViewCompositionalLayout.list(using: configuration)
+    }()
+    
+    private static let snapshot: NSDiffableDataSourceSnapshot<MainViewModel.AppSection, String> = {
+        var snapshot = NSDiffableDataSourceSnapshot<MainViewModel.AppSection, String>()
+        snapshot.appendSections(MainViewModel.AppSection.allCases)
+        MainViewModel.AppSection.allCases.forEach {
+            snapshot.appendItems([$0.navigationTitle], toSection: $0)
+        }
+        return snapshot
+    }()
+    
+    private lazy var dataSource: UICollectionViewDiffableDataSource<MainViewModel.AppSection, String> = {
+        let registration = UICollectionView.CellRegistration<UICollectionViewListCell, String> { (cell, indexPath, string) in
+            var content = cell.defaultContentConfiguration()
+            content.attributedText = NSAttributedString(string, style: .title)
+            cell.contentConfiguration = content
+        }
+        
+        return UICollectionViewDiffableDataSource<MainViewModel.AppSection, String>(
+            collectionView: collectionView
+        ) { (collectionView, indexPath, string) -> UICollectionViewCell? in
+            let cell = collectionView.dequeueConfiguredReusableCell(
+                using: registration,
+                for: indexPath,
+                item: string
+            )
+            
+            let primaryColor = UIColor(.ui.grey6)
+            
+            let backgroundView = UIView()
+            backgroundView.backgroundColor = primaryColor.withAlphaComponent(0.2)
+            cell.backgroundView = backgroundView
+            
+            let selectedBackgroundView = UIView()
+            selectedBackgroundView.backgroundColor = primaryColor
+            cell.selectedBackgroundView = selectedBackgroundView
+            
+            return cell
+        }
+    }()
+    
+    private lazy var collectionView: UICollectionView = UICollectionView(
+        frame: .zero,
+        collectionViewLayout: Self.layout
+    )
+    private let model: MainViewModel
+    private var subscriptions: Set<AnyCancellable> = []
+
+    init(model: MainViewModel) {
+        self.model = model
+        super.init(nibName: nil, bundle: nil)
+        
+        self.title = "Pocket"
+        
+        collectionView.alwaysBounceVertical = true
+        collectionView.delegate = self
+        collectionView.dataSource = dataSource
+        dataSource.apply(Self.snapshot)
+        
+        model.$selectedSection
+            .receive(on: DispatchQueue.main)
+            .sink { [weak self] section in
+                let indexPath = self?.dataSource.indexPath(for: section.navigationTitle)
+                self?.collectionView.selectItem(at: indexPath, animated: false, scrollPosition: .top)
```

Comment:
1. I'd suggest pulling this bit of code into a private function so we don't have to unwrap self a bunch of times. I get that it's only two lines, but still. It also helps avoid passing optionals into the `collectionView.selectItem(at:)` method.
2. See not above regarding building the snapshot. Getting the indexPath could be something like:
```
let indexPath: IndexPath = [0, MainViewModel.AppSection.allCases.index(of: section)]
```